### ISSUE TITLE
Question about optimizer in pytorch ddp training

### ISSUE THREAD

**Issue by ForeverFancy** [rosinality/stylegan2-pytorch#245]
Thanks for the great work. 

Follow the link of [pytorch ddp training tutorial](https://pytorch.org/tutorials/intermediate/ddp_tutorial.html), they set distributed data parallel model's parameters in the optimizer like:

```python
ddp_model = DDP(model, device_ids=[rank])
optimizer = optim.SGD(ddp_model.parameters(), lr=0.001)
```

However, in this implementation, the **original model's parameters** (not ddp model) are set in the optimizer like:

```python
generator = Generator()
g_optim = optim.Adam(
        generator.parameters(),
        lr=args.lr * g_reg_ratio,
        betas=(0 ** g_reg_ratio, 0.99 ** g_reg_ratio),
    )
generator = nn.parallel.DistributedDataParallel(
            generator,
            device_ids=[args.local_rank],
            output_device=args.local_rank,
            broadcast_buffers=False,
        )
```

So I wonder whether your implementation would influence the sync of gradients between different GPUs, or whether it is the same as the first implementation?

Thanks!

**Comment by rosinality** [comment]
Yes, it would be same as sync occurs during backward with ddp.

**Comment by ForeverFancy** [comment]
> Yes, it would be same as sync occurs during backward with ddp.

OK, thanks.
